[qt5-location/CI] hash failure for patch

```
Building package qt5-location[core]:x64-windows-static...
Could not locate cached archive: W:\0c\0c35a445de1e9991e1cfcce53746a5ecdb7d094a.zip
-- Downloading https://codereview.qt-project.org/gitweb?p=qt%2Fqtlocation.git;a=commitdiff_plain;h=6b2cf7e9d150b7be709fcd688c5045949cedc3d9;hp=7769ea903f87efc4ad55530a2749f104eddff2e4...
CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:103 (message):
  

  File does not have expected hash:

          File path: [ D:/downloads/temp/qt5-location-rename-99d16fb0.patch ]
      Expected hash: [ 99d16fb0e88a2250de3896815abbb22ff5aa4d3920397610cf37be701fe03a7241e0586aae5b85755aeb958926183c96a0482a8837335d20a2171ebb2a66e640 ]
        Actual hash: [ 68c25752b7743e43fc2a7a9144ae354ebaf9bacf4ec6e81c8f5ed31c7fd505103adcd57d97a009c0a8de2d586a6bec50aa30ea4c100c5a96ea6acf881db7475a ]

  The file may have been corrupted in transit.  This can be caused by
  proxies.  If you use a proxy, please set the HTTPS_PROXY and HTTP_PROXY
  environment variables to
  "https://user:***@your-proxy-ip-address:port/".

  

Call Stack (most recent call first):
  scripts/cmake/vcpkg_download_distfile.cmake:189 (test_hash)
  ports/qt5-location/portfile.cmake:1 (vcpkg_download_distfile)
  scripts/ports.cmake:136 (include)


Error: Building package qt5-location:x64-windows-static failed with: BUILD_FAILED

```

Hi @Neumann-A 

Thanks for posting this issue.

It seems that the hash value for the patch is varying.

```
Starting package 225/227: qt5-location:x86-windows
Building package qt5-location[core]:x86-windows...
-- Downloading https://codereview.qt-project.org/gitweb?p=qt%2Fqtlocation.git;a=commitdiff_plain;h=6b2cf7e9d150b7be709fcd688c5045949cedc3d9;hp=7769ea903f87efc4ad55530a2749f104eddff2e4...
CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:103 (message):
  

  File does not have expected hash:

          File path: [ D:/downloads/temp/qt5-location-rename-99d16fb0.patch ]
      Expected hash: [ 99d16fb0e88a2250de3896815abbb22ff5aa4d3920397610cf37be701fe03a7241e0586aae5b85755aeb958926183c96a0482a8837335d20a2171ebb2a66e640 ]
        Actual hash: [ 1ccbdb8d843405bee6df935f2dfb9f15aaee3865cab39d314d1bc74269db32071cbd014f7036c8ce5c5b84dea33456a57e31c005a1c72c76ea5db7fe392ed762 ]
```
https://dev.azure.com/vcpkg/public/_build/results?buildId=45549&view=logs&j=5dffab7c-3299-5a76-7b02-e037cd868fad&t=1294d155-2ef8-5412-c4b1-d5b95781dd72&l=3552

We need to continue to investigate it.

--- a/ports/qt5-location/CONTROL
+++ b/ports/qt5-location/CONTROL
@@ -1,5 +1,5 @@
 Source: qt5-location
 Version: 5.15.1
-Port-Version: 1
+Port-Version: 2
 Description: Qt5 Location Module - Displays map, navigation, and place content in a QML application.
 Build-Depends: qt5-base[core], qt5-declarative, qt5-quickcontrols, qt5-quickcontrols2, qt5-serialport

--- a/ports/qt5-location/portfile.cmake
+++ b/ports/qt5-location/portfile.cmake
@@ -1,10 +1,3 @@
-vcpkg_download_distfile(
-    PATCH
-    URLS "https://codereview.qt-project.org/gitweb?p=qt%2Fqtlocation.git;a=commitdiff_plain;h=6b2cf7e9d150b7be709fcd688c5045949cedc3d9;hp=7769ea903f87efc4ad55530a2749f104eddff2e4"
-    SHA512 99d16fb0e88a2250de3896815abbb22ff5aa4d3920397610cf37be701fe03a7241e0586aae5b85755aeb958926183c96a0482a8837335d20a2171ebb2a66e640
-    FILENAME qt5-location-rename-99d16fb0.patch
-)
-
 message(STATUS "${PORT} has a spurious failure in which it is unable to create a parent directory! Just retry.")
 include(${CURRENT_INSTALLED_DIR}/share/qt5/qt_port_functions.cmake)
-qt_submodule_installation(PATCHES "${PATCH}")
+qt_submodule_installation(PATCHES rename_third_party_libs_conflicts.patch)

--- a/ports/qt5-location/rename_third_party_libs_conflicts.patch
+++ b/ports/qt5-location/rename_third_party_libs_conflicts.patch
@@ -1,0 +1,68 @@
+diff --git a/src/3rdparty/clip2tri/clip2tri.pro b/src/3rdparty/clip2tri/clip2tri.pro
+index 802c040..4059a63 100644
+--- a/src/3rdparty/clip2tri/clip2tri.pro
++++ b/src/3rdparty/clip2tri/clip2tri.pro
+@@ -1,4 +1,4 @@
+-TARGET = clip2tri
++TARGET = qt_clip2tri
+ 
+ CONFIG += staticlib exceptions warn_off optimize_full
+ 
+@@ -18,5 +18,5 @@ gcc {
+ HEADERS += clip2tri.h
+ SOURCES += clip2tri.cpp
+ 
+-LIBS_PRIVATE += -L$$MODULE_BASE_OUTDIR/lib -lpoly2tri$$qtPlatformTargetSuffix() -lclipper$$qtPlatformTargetSuffix()
++LIBS_PRIVATE += -L$$MODULE_BASE_OUTDIR/lib -lqt_poly2tri$$qtPlatformTargetSuffix() -lqt_clipper$$qtPlatformTargetSuffix()
+ 
+diff --git a/src/3rdparty/clipper/clipper.pro b/src/3rdparty/clipper/clipper.pro
+index 874d55c..a518d24 100644
+--- a/src/3rdparty/clipper/clipper.pro
++++ b/src/3rdparty/clipper/clipper.pro
+@@ -1,4 +1,4 @@
+-TARGET = clipper
++TARGET = qt_clipper
+ 
+ CONFIG += staticlib exceptions warn_off optimize_full
+ 
+diff --git a/src/3rdparty/poly2tri/poly2tri.pro b/src/3rdparty/poly2tri/poly2tri.pro
+index 76f2779..6c5f769 100644
+--- a/src/3rdparty/poly2tri/poly2tri.pro
++++ b/src/3rdparty/poly2tri/poly2tri.pro
+@@ -1,4 +1,4 @@
+-TARGET = poly2tri
++TARGET = qt_poly2tri
+ 
+ CONFIG += staticlib warn_off optimize_full
+ 
+diff --git a/src/location/declarativemaps/declarativemaps.pri b/src/location/declarativemaps/declarativemaps.pri
+index e2a922f..12199ee 100644
+--- a/src/location/declarativemaps/declarativemaps.pri
++++ b/src/location/declarativemaps/declarativemaps.pri
+@@ -68,5 +68,5 @@ SOURCES += \
+         declarativemaps/qquickgeomapgesturearea.cpp
+ 
+ load(qt_build_paths)
+-LIBS_PRIVATE += -L$$MODULE_BASE_OUTDIR/lib -lpoly2tri$$qtPlatformTargetSuffix() -lclip2tri$$qtPlatformTargetSuffix()
++LIBS_PRIVATE += -L$$MODULE_BASE_OUTDIR/lib -lqt_poly2tri$$qtPlatformTargetSuffix() -lqt_clip2tri$$qtPlatformTargetSuffix()
+ 
+diff --git a/src/location/location.pro b/src/location/location.pro
+index 1b541b9..b0e2c3f 100644
+--- a/src/location/location.pro
++++ b/src/location/location.pro
+@@ -45,4 +45,4 @@ HEADERS += $$PUBLIC_HEADERS $$PRIVATE_HEADERS
+ 
+ load(qt_module)
+ 
+-LIBS_PRIVATE += -L$$MODULE_BASE_OUTDIR/lib -lclip2tri$$qtPlatformTargetSuffix()
++LIBS_PRIVATE += -L$$MODULE_BASE_OUTDIR/lib -lqt_clip2tri$$qtPlatformTargetSuffix()
+diff --git a/src/positioning/positioning.pro b/src/positioning/positioning.pro
+index 6535090..db49921 100644
+--- a/src/positioning/positioning.pro
++++ b/src/positioning/positioning.pro
+@@ -95,4 +95,4 @@ HEADERS += $$PUBLIC_HEADERS $$PRIVATE_HEADERS
+ 
+ load(qt_module)
+ 
+-LIBS_PRIVATE += -L$$MODULE_BASE_OUTDIR/lib -lclip2tri$$qtPlatformTargetSuffix()
++LIBS_PRIVATE += -L$$MODULE_BASE_OUTDIR/lib -lqt_clip2tri$$qtPlatformTargetSuffix()

